t/ui/27-plugin_obs_rsync_obs_status.t

```diff
@@ -71,28 +82,87 @@ my %fake_response_by_project = (
-                my $c = shift;
-                my $pkg = $c->param('package');
-                return $c->render(status => 404) if !$pkg and $project ne 'Proj1';
-                return $c->render(status => 200, text => $fake_response_by_project{$project});
-            });
+
+my $server_process = sub {
+    my $mock = Mojolicious->new;    # uncoverable statement
+    $mock->mode('test');    # uncoverable statement
+
+    my $www_authenticate = qq(Signature realm="Use your developer account",headers="(created)"); # uncoverable statement
+    $mock->routes->get(
+        '/build/ProjWithAuth/_result' => sub {
+            my $c = shift;    # uncoverable statement
+
+            if ($c->req->headers->authorization) {    # uncoverable statement
+                return $c->render(status => 200, text => $fake_response_by_project{Proj1});    # uncoverable statement
+            }
+
+            $c->res->headers->www_authenticate($www_authenticate);    # uncoverable statement
+            ;    # uncoverable statement
+            return $c->render(status => 401, text => 'login');    # uncoverable statement
+        });    # uncoverable statement
+
+    $mock->routes->get(
+        '/build/ProjTestingSignature/_result' => sub {    # uncoverable statement
+            my $c = shift;    # uncoverable statement
+
+            my $client_auth_header = $c->req->headers->authorization // '';    # uncoverable statement
+
+            if ($auth_header_exact eq $client_auth_header) {    # uncoverable statement
+
+                return $c->render(status => 200, text => $fake_response_by_project{Proj1});    # uncoverable statement
+            }    # uncoverable statement
+
+            $c->res->headers->www_authenticate($www_authenticate);    # uncoverable statement
+            return $c->render(status => 401, text => 'login');    # uncoverable statement
+        });    # uncoverable statement
+
+    for my $project (sort keys %fake_response_by_project) {    # uncoverable statement
+        $mock->routes->get(    # uncoverable statement
+            "/build/$project/_result" => sub {    # uncoverable statement
+                my $c = shift;    # uncoverable statement
+                my $pkg = $c->param('package');    # uncoverable statement
+                return $c->render(status => 404) if !$pkg and $project ne 'Proj1';    # uncoverable statement
+                return $c->render(status => 200, text => $fake_response_by_project{$project});   # uncoverable statement
+            });    # uncoverable statement
     }
-    my $daemon = Mojo::Server::Daemon->new(app => $mock, listen => [$host]);
-    $daemon->run;
-    note 'Fake API server stopped';
-    _exit(0);
+    my $daemon = Mojo::Server::Daemon->new(app => $mock, listen => [$host]);    # uncoverable statement
+    $daemon->run;    # uncoverable statement
+    note 'Fake API server stopped';    # uncoverable statement
+    _exit(0);    # uncoverable statement
```

Comment:
To get coverage working, add these lines:
```
    # server process
    Devel::Cover::report() if Devel::Cover->can('report');
    _exit(0);    # uncoverable statement
  },
  max_kill_attempts => 0,
  blocking_stop => 1,
  _default_blocking_signal => POSIX::SIGTERM,
  kill_sleeptime => 0;
```
Then you should be able to remove all the other "uncoverable" markers.

Reply:
Just applied it and locally the report is 100% coverage! :tada: 

This looks **MUCH** cleaner than the markers all over the place